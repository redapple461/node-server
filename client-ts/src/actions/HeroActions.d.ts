export declare const SET_MARVEL = "SET_MARVEL";
export declare const SET_DC = "SET_DC";
export declare const SET_BOTH = "SET_BOTH";
export declare const GET_DATA = "GET_DATA";
export declare const HEROES_LOADED = "HEROES_LOADED";
export declare const ADD_HERO = "ADD_HERO";
export declare const DELETE_HERO = "DELETE_HERO";
export declare const INIT_HERO = "INIT_HERO";
export declare const UPDATE_NAME = "UPDATE_NAME";
export declare const UPDATE_ADD_HERO_NAME = "UPDATE_ADD_HERO_NAME";
export declare const UPDATE_ADD_HERO_UNIVERSE = "UPDATE_ADD_HERO_UNIVERSE";
export declare const CLEAR_DETAIL_HERO = "CLEAR_DETAIL_HERO";
export declare const DETAIL_LOADED = "DETAIL_LOADED";
export declare const UPDATE_HERO = "UPDATE_HERO";
export declare const UPDATE_UNIVERSE = "UPDATE_UNIVERSE";
export declare const ADD_ASKILL = "ADD_ASKILL";
export declare const REMOVE_ASKILL = "REMOVE_ASKILL";
export declare const CLEAR_ADD_HERO = "CLEAR_ADD_HERO";
export declare const SET_EMAIL = "SET_EMAIL";
export declare const SET_PASSWORD = "SET_PASSWORD";
export declare const SET_REG_EMAIL = "SET_REG_EMAIL";
export declare const SET_REG_PASSWORD = "SET_REG_PASSWORD";
export declare const SET_USER_ID = "SET_USER_ID";
export declare const SET_JWT = "SET_JWT";
export declare const LOGOUT = "LOGOUT";
export declare const SET_USER_NAME = "SET_USER_NAME";
export declare const SET_USER_SURNAME = "SET_USER_SURNAME";
export declare const SET_USER_PHONE = "SET_USER_PHONE";
export declare const CLEAR_REG_DATA = "CLEAR_REG_DATA";
