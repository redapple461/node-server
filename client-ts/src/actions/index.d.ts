import { Hero } from '../models/Hero';
import { HeroActionsType } from '../actions/types/HeroTypes';
export declare function setMarvel(): HeroActionsType;
export declare function setDC(): HeroActionsType;
export declare function setBoth(): HeroActionsType;
export declare function getData(data: Hero[]): HeroActionsType;
export declare function loadComplete(): HeroActionsType;
export declare function addHero(hero: Hero): HeroActionsType;
export declare function deleteHero(id: number): HeroActionsType;
export declare function initDetailHero(hero: Hero): HeroActionsType;
export declare function updateName(newName: string): HeroActionsType;
export declare function updateUniverse(newUniverse: string): HeroActionsType;
export declare function updateHero(): HeroActionsType;
export declare function updateAddHeroName(addName: string): HeroActionsType;
export declare function updateAddHeroUniverse(addUniverse: string): HeroActionsType;
export declare function clearDetailHero(): HeroActionsType;
export declare function isLoad(): HeroActionsType;
export declare function addSkill(skillToAdd: string): HeroActionsType;
export declare function removeSkill(skillToRemove: string): HeroActionsType;
export declare function clearAddHero(): {
    type: string;
};
export declare function setEmail(email: string): HeroActionsType;
export declare function setPassword(password: string): HeroActionsType;
export declare function setRegEmail(regEmail: string): HeroActionsType;
export declare function setRegPassword(regPassword: string): HeroActionsType;
export declare function setUserId(userId: string): HeroActionsType;
export declare function setJWT(jwt: string): HeroActionsType;
export declare function logout(): HeroActionsType;
export declare function setUserName(userName: string): HeroActionsType;
export declare function setUserSurname(surname: string): HeroActionsType;
export declare function setUserPhone(phone: string): HeroActionsType;
export declare function clearRegData(): HeroActionsType;
