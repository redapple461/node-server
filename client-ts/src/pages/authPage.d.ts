/// <reference types="react" />
import './css/authPage.css';
export declare const AuthPage: () => JSX.Element;
