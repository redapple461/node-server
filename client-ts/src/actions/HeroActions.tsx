export const SET_MARVEL = 'SET_MARVEL';
export const SET_DC = 'SET_DC';
export const SET_BOTH = 'SET_BOTH';
export const GET_DATA = 'GET_DATA';
export const HEROES_LOADED = 'HEROES_LOADED';
export const ADD_HERO = 'ADD_HERO';
export const DELETE_HERO = 'DELETE_HERO';
export const INIT_HERO = 'INIT_HERO';
export const UPDATE_NAME = 'UPDATE_NAME';
export const UPDATE_ADD_HERO_NAME = 'UPDATE_ADD_HERO_NAME';
export const UPDATE_ADD_HERO_UNIVERSE = 'UPDATE_ADD_HERO_UNIVERSE';
export const CLEAR_DETAIL_HERO = 'CLEAR_DETAIL_HERO';
export const DETAIL_LOADED = 'DETAIL_LOADED';
export const UPDATE_HERO = 'UPDATE_HERO';
export const UPDATE_UNIVERSE = 'UPDATE_UNIVERSE';
