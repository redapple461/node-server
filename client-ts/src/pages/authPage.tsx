import React, { useEffect } from 'react';
import './css/authPage.css';
import { Button } from '../components/button';
import { useDispatch, useSelector } from 'react-redux';
import * as actions from '../actions/index';
import { HeroActionsType } from '../actions/types/HeroTypes';
import * as http from '../http/httpHook';
import { HeroStore } from '../interfaces/iStore/HeroStore';
import { Link } from 'react-router-dom';

export const AuthPage = () => {
	const email = useSelector((state: HeroStore) => state.email);
	const password = useSelector((state: HeroStore) => state.password);
	const storageName = 'userData';
	const dispatch = useDispatch();
	let timeout = null;

	useEffect(() => {
		const data = JSON.parse(localStorage.getItem(storageName))

		if (data && data.token) {
			dispatch(actions.setJWT(data.token));
			dispatch(actions.setUserId(data.userId));
		};
	});

	const inputEventHandler = (e: React.ChangeEvent<HTMLInputElement>, action: HeroActionsType) => {
		e.persist();
		clearTimeout(timeout);
		timeout = setTimeout(() => {
			dispatch(action);
		}, 300);
	};

	const buttonEventHandler = () => {
		http.signIn(email, password)
		.then(res => {
			if (res.message) {
				return window.M.toast({html: res.message});
			}
			window.M.toast({html: `Hello ${res.name}`});
			localStorage.setItem(storageName, JSON.stringify({
				userId: res.userId,
				token: res.token
			}));
			dispatch(actions.setJWT(res.token));
			dispatch(actions.setUserId(res.userId));
		});
	};

	return(
			<div className='row'>
				<div className='col s6 offset-s3'>
					<h1> Auth page </h1>
					<div className='card teal lighten-1'>
						<div className='card-content white-test'>
							<div>
								<div className='input-field'>
									<input placeholder='Enter email' id='email' type='text' onChange={(e) => inputEventHandler(e, actions.setEmail(e.target.value))}/>
									<label htmlFor='email'>Email</label>
								</div>
								<div className='input-field'>
									<input placeholder='Enter password' id='password' type='password' onChange={(e) => inputEventHandler(e, actions.setPassword(e.target.value))}/>
									<label htmlFor='password'>Password</label>
								</div>
							</div>
							<div className='card-action'>
								<Button className='waves-effect waves-light btn yellow darken-4' text='Sign in' onClick={buttonEventHandler}/>
								<Link to={{pathname: '/register'}} >
									<Button className='waves-effect waves-light btn grey lighten-1' text='Register' />
								</Link>
							</div>
						</div>
					</div>
				</div>
			</div>
	);
}
