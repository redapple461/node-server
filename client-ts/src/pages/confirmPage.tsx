import React from 'react';
import { Button } from '../components/button';
import { changePassword } from '../http/httpHook';
import bcrytp from 'bcryptjs';
import { useDispatch, useSelector } from 'react-redux';
import { redirectToResponse } from '../actions';
import { HeroStore } from '../interfaces/iStore/HeroStore';
import { Redirect } from 'react-router-dom';

export const ConfirmPage = (props) => {
	let firstPassword: string = '';
	let confirmPassword: string = '';
	const dispatch = useDispatch();
	const fireRedirect = useSelector( (store: HeroStore) => store.redirect);
	const change = () => {
		if (firstPassword === '' || confirmPassword === '') {
			return alert('Missed password');
		}
		if (firstPassword !== confirmPassword) {
			return alert('Password not equal');
		} else {
			if (firstPassword.length < 7) {
				return alert('Password length is less then 7');
			}
			bcrytp.hash(firstPassword, 5).then(_res =>
				changePassword(_res, props.match.params.token).then(res => {
					if (res.error) {
						window.M.toast({html: res.error});
					} else {
						window.M.toast({html: 'Password changed'});
						dispatch(redirectToResponse());
					}
				})
			);
		}
	};

	return(
		<>
			<div className='row'>
				<div className='col s6 offset-s3'>
					<div className='card teal lighten-4'>
						<div className='card-title'>
							Type your new password
						</div>
						<div className='card-content'>
							<div className='input-field'>
								<input id='first-password' type='password' onChange={(e) => firstPassword = e.target.value}/>
								<label htmlFor='first-password'>Type new password</label>
							</div>
							<div className='input-field'>
								<input id='second-password' type='password' onChange={(e) => confirmPassword = e.target.value}/>
								<label htmlFor='second-password'>Confrim password</label>
							</div>
						</div>
						<div className='card-actions'>
						<Button className='waves-effect waves-light btn yellow darken-4' text='Change password' onClick={() => change() }/>
						</div>
					</div>
				</div>
				{fireRedirect && (
					<Redirect to={{pathname: '/'}}/>
				)}
			</div>
		</>
	);
};
