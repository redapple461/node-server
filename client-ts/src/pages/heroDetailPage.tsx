import React, { useEffect } from 'react';
import './css/heroDetail.css';
import { useSelector, useDispatch } from 'react-redux';
import * as actions from '../actions';
import { NoPage } from '../components/404';
import {  Details } from '../components/detailPage';
import { RadioButton } from '../components/radio';
import { Button } from '../components/button';
import { getByName } from '../http/httpHook';
import { HeroStore } from '../interfaces/iStore/HeroStore';
import { updateHero } from '../http/httpHook';

export const HeroDetailPage = (props: any) => {
	const token = useSelector((state: HeroStore) => state.jwt);
	const oldname = useSelector((state: HeroStore) => state.oldName);
	const detailHero = useSelector((state: HeroStore) => state.detailHero);
	const dispatch = useDispatch();
	const isLoad = useSelector((state: HeroStore) => state.isLoad);
	async function fetchData () {
		try {
			await getByName(props.match.params.name, token)
			.then(res => {
				if (res.message) {
					return window.M.toast({html: 'No auth!!!'});
				}
				dispatch(actions.initDetailHero(res));
		});
		} catch (e) {
			// tslint:disable-next-line: no-console
			console.log(e);
		}
		}
	const goBack = () => {
		props.history.goBack();
	};

	useEffect(() => {
			if (isLoad === false) {
				fetchData();
				dispatch(actions.isLoad());
			}
	});

	if (!detailHero) {
		return (
			<NoPage name={props.match.params.name} />
		);
	}
	const logout = () => {
		dispatch(actions.logout());
		localStorage.removeItem('userData');
	};
	//               onChange={(e) => dispatch(actions.updateUniverse(e.target.value))}
	return(
		<>
				<Button className='waves-effect waves-light btn rightbtn' text='Logout' onClick={() => logout()}/>
				<Details name={detailHero.name} id={detailHero.id} universe={detailHero.universe} skills={detailHero.skills}/>
				<div className='form'>
					<p> Type new name of hero and choose his universe</p>
					<input
						type='text'
						placeholder={detailHero.name}
						onChange={(e) => {dispatch(actions.updateName(e.target.value)); }}
					/>
					<div>
						<RadioButton
								className='with-gap'
								value='Marvel'
								text='Marvel'
								checked={detailHero.universe === 'Marvel'}
								dispatch={() =>  dispatch(actions.updateUniverse('Marvel'))}
							/>
						<RadioButton
							className='with-gap'
							value='DC'
							text='DC'
							checked={detailHero.universe === 'DC'}
							dispatch={() =>  dispatch(actions.updateUniverse('DC'))}
						/>
					</div>
				</div>

				<Button
					className='waves-effect waves-light btn'
					type='button'
					onClick={() => {goBack(); }}
					text='Back'
				/>
				<Button
					className='waves-effect waves-light btn'
					type='button'
					onClick={() => { updateHero(oldname, detailHero, token).then(() => dispatch(actions.updateHero()))} }
					text='Save'
				/>
			</>
	);
};
