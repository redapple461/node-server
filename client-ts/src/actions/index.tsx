import { Hero } from '../models/Hero';
import {HeroActionsType} from '../actions/types/HeroTypes';
import * as actions from '../actions/HeroActions';

export function setMarvel (): HeroActionsType {
	return{
		type: actions.SET_MARVEL
	};
}

export  function setDC (): HeroActionsType {
	return{
		type: actions.SET_DC
	};
}

export function setBoth (): HeroActionsType{
	return{
		type: actions.SET_BOTH
	};
}

export function getData (data: Hero[]): HeroActionsType {
	return{
		type: actions.GET_DATA,
		data: data
	};
}

export function loadComplete(): HeroActionsType {
	return{
		type: actions.HEROES_LOADED
	};
}

export function addHero(hero: Hero): HeroActionsType {
	return{
		type: actions.ADD_HERO,
		hero: hero
	};
}

export function deleteHero(id: number): HeroActionsType{
	return{
		type: actions.DELETE_HERO,
		id: id
	};
}

export function initDetailHero(hero: Hero): HeroActionsType {
	return{
		type: actions.INIT_HERO,
		hero: hero
	};
}

export function updateName(name: string): HeroActionsType {
	return{
		type: actions.UPDATE_NAME,
		newName: name
	};
}

export function updateUniverse(universe: string): HeroActionsType{
	return{
		type: actions.UPDATE_UNIVERSE,
		newUniverse: universe
	};
}

export function updateHero(): HeroActionsType {
	return{
		type: actions.UPDATE_HERO
	};
}

export function updateAddHeroName(name: string): HeroActionsType{
	return{
		type: actions.UPDATE_ADD_HERO_NAME,
		addName: name,
	};
}

export function updateAddHeroUniverse(universe: string): HeroActionsType{
	return{
		type: actions.UPDATE_ADD_HERO_UNIVERSE,
		addUniverse: universe
	};
}

export function clearDetailHero(): HeroActionsType{
	return{
		type: actions.CLEAR_DETAIL_HERO
	};
}

export function isLoad(): HeroActionsType {
	return{
		type: actions.DETAIL_LOADED
	};
}
