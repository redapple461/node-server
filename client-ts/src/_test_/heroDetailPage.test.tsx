import React from 'react';
import {HeroDetailPage} from '../pages/heroDetailPage';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Switch, Route } from 'react-router-dom';
import {BrowserRouter as Router} from 'react-router-dom';
import * as actions from '../actions';
import 'babel-polyfill'
import jest from 'jest-mock';

describe('<HeroList/> test', () => {
	let store;
	let container;
	const mockStore = configureStore([]);

	// fetch = jest.fn(() => Promise.resolve());

	beforeEach(() => {
		store = mockStore({
			searchUniverse: '',
			heroes: [],
			oldName: '',
			addHero: {name: '', universe: '', skills: []},
			detailHero: {id: 0, name: 'Ironman', universe: 'DC', skills: ['Heal']},
			isLoad: false,
			noHeroes: true
		});
		store.dispatch = jest.fn();

		container = renderer.create(
			<Provider store={store}>
				<Router>
				<Route>
					<HeroDetailPage history={{goBack : jest.fn()}} match={{params: 'no_hero'}} />
				</Route>
				</Router>
			</Provider>
		);

	});

	it('should render with given state from Redux store', done => {
		expect(container.toJSON()).toMatchSnapshot();
		done();
	});

	it('shouldnt call dispatch at start', done => {
		expect(store.dispatch).toHaveBeenCalledTimes(0);
		done();
	});

	it('should call dispatch on input change', done => {
		renderer.act(() => {
			container.toJSON()[8].children[1].props.onChange({ target: { value: 'Mock_2' } });
		});
		expect(store.dispatch).toHaveBeenCalledTimes(2);
		expect(store.dispatch).toHaveBeenCalledWith({
				type: 'UPDATE_NAME',
				newName: 'Mock_2'
			}
		);
		done();
	});

	it('should call dispatch on radio btn change (Marvel)', done => {
		renderer.act(() => {
			container.toJSON()[8].children[2].children[0].children[0].props.onChange();
		});
		expect(store.dispatch).toHaveBeenCalledTimes(2);
		expect(store.dispatch).toHaveBeenCalledWith(
			actions.updateUniverse('Marvel')
		);
		done();
	});

	it('should call dispatch on radio btn change (DC)', done => {
		renderer.act(() => {
			container.toJSON()[8].children[2].children[1].children[0].props.onChange();
		});
		expect(store.dispatch).toHaveBeenCalledTimes(2);
		expect(store.dispatch).toHaveBeenCalledWith(
			actions.updateUniverse('DC')
		);
		done();
	});

	it('should call dispatch on save btn click', done => {
		renderer.act(() => {
			container.toJSON()[10].props.onClick();
		});
		expect(store.dispatch).toHaveBeenCalledTimes(2);
		done();
	});

});