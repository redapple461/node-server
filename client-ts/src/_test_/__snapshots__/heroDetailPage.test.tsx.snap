// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroList/> test should render with given state from Redux store 1`] = `
Array [
  <h1>
    Ironman
     details
  </h1>,
  <strong>
     id: 
    0
     
  </strong>,
  <br />,
  <strong>
     Name: 
    Ironman
     
  </strong>,
  <br />,
  <strong>
     Universe: 
    DC
     
  </strong>,
  <br />,
  <strong>
     Skills : 
    <ul
      style={
        Object {
          "color": "red",
        }
      }
    >
      <div>
         
        Heal
         
      </div>
    </ul>
  </strong>,
  <div
    className="form"
  >
    <p>
       Type new name of hero and choose his universe
    </p>
    <input
      onChange={[Function]}
      placeholder="Ironman"
      type="text"
    />
    <div>
      <label>
        <input
          checked={false}
          className="with-gap"
          onChange={[Function]}
          type="radio"
          value="Marvel"
        />
        <span>
          Marvel
        </span>
      </label>
      <label>
        <input
          checked={true}
          className="with-gap"
          onChange={[Function]}
          type="radio"
          value="DC"
        />
        <span>
          DC
        </span>
      </label>
    </div>
  </div>,
  <button
    className="waves-effect waves-light btn"
    onClick={[Function]}
  >
     
    Back
     
  </button>,
  <button
    className="waves-effect waves-light btn"
    onClick={[Function]}
  >
     
    Save
     
  </button>,
]
`;
